### ISSUE TITLE
Data tab `SELECT` query blocks main thread

### ISSUE THREAD

**Issue by geemark** [HeidiSQL/HeidiSQL#1694]
**Preconditions**

* HeidiSQL version: 12.1.0.6564 (64bit)
* Database type and version: MySQL 8
* OS: Windows 11 Pro

**Describe the bug**
When clicking the "Data" tab after having selected a table, it appears that it will re-run the previously used query. This is fine, but there is a major issue with this, in that it seems to be doing this on the UI thread. Therefore the entire program becomes unresponsive until that query finishes, and there is no way to cancel it, because the UI freezes while doing this.

This is problematic if the query that was last used on this table is very heavy, which will either lead the program to a crash or force you to wait a very long time for a query you often do not need anyway.

The same issue occurs if you just change the filter query and "Apply filter". If you accidentally create a heavy query, you can either wait for it to finish (wait for the UI to respond again), or force crash the program and start over - and avoid the "Data" tab next time, since that will have remembered the filter and auto-execute it and cause the same issue again.

**To Reproduce**
1. Enter a (heavy) query in the "Data" tab ->filters section
2. Go to a different table
3. Go back to the same table and click "Data"
4. Program is unresponsive until the old query is re-executed

**Screenshots or Crash reports**

N/A

**Comment by ansgarbecker** [comment]
Please use the "Clear data tab filter" menu item before activating the data tab:
https://www.heidisql.com/forum.php?t=17107

Alternatively, you can tell HeidiSQL not to restore filters you used in a prior session:
![grafik](https://user-images.githubusercontent.com/7986591/199306608-c0df6974-9123-488c-841c-a43cd0446aff.png)

**Comment by geemark** [comment]
Thank you for your suggestions. However, I would still like to keep the filters in most cases, since I think it is a very nice feature. Simply disabling the feature will not fix the issue with the feature :)

This also only "solves" the first issue. I still do believe it is a bug to run SQL queries on the UI thread, since that will cause the UI to hang (and eventually crash) on long-running queries.

**Comment by FingerlessGlov3s** [comment]
I hit this problem every now and then, but it would be good if it didn't lockup the UI thread, and you could cancel the filter query.

Sometimes I'm using the data tab, and then write a filter, which could take 60 seconds to query the database, but I'd rather cancel the query and then write it differently to speed it up or I've noticed it's wrong. At the moment I just have to wait, it would be great if the query ran in it's own thread and you could cancel it.

If it could run in it's own thread would be good if the query time was recorded as well in the log at the bottom, like it does when you run a query a query tab.

**Comment by mertkarakulak** [comment]
I don't think this solves the problem; it only eliminates one of the contributing factors. The real issue here is preventing the main thread from freezing due to any issue.

> Please use the "Clear data tab filter" menu item before activating the data tab: https://www.heidisql.com/forum.php?t=17107
> 
> Alternatively, you can tell HeidiSQL not to restore filters you used in a prior session: ![grafik](https://user-images.githubusercontent.com/7986591/199306608-c0df6974-9123-488c-841c-a43cd0446aff.png)

**Comment by ansgarbecker** [comment]
The issue is still open and accepted, the above is just a recommendation or a workaround.

**Comment by FingerlessGlov3s** [comment]
Would love to see a way to cancel a filter lookup, as can you apply a filter, realise actually it's going to take an age to return, and you either have to kill the process or login to the db itself, and kill the query.

Would be nice to make this more graceful
